ipython/dns/UI.md

```diff
@@ -0,0 +1,592 @@
+          + [FilterInput.react.js](#filterinputreactjs)
+          + [GridPanelMixin.react.js](#gridpanelmixinreactjs)
+          + [IncidentProgressionPanel.react.js](#incidentprogressionpanelreactjs)
+          + [IPythonNotebookPanel.react.js](#ipythonnotebookpanelreactjs)
+          + [NetworkPanel.react.js](#networkpanelreactjs)
+          + [Panel.react.js](#panelreactjs)
+          + [SuspiciousPanel.react.js](#suspiciouspanelreactjs)
+        * [Constants](#constants)
+          + [DnsConstants.js](#dnsconstantsjs)
+        * [Dispatchers](#dispatchers)
+          + [DnsDispatcher.js](#dnsdispatcherjs)
+        * [Stores](#stores)
+          + [CommentsStore.js](#commentsstorejs)
+          + [DendrogramStore.js](#dendrogramstorejs)
+          + [DetailsStore.js](#detailsstorejs)
+          + [DnsAppStore.js](#dnsappstorejs)
+          + [IncidentProgressionStore.js](#incidentprogressionstorejs)
+          + [RestStore.js](#reststorejs)
+          + [SuspiciousStore.js](#suspiciousstorejs)
+      - [Dev Requirements](#dev-requirements)
+      - [Development/Debugging process](#developmentdebugging-process)
+      - [Building modules](#building-modules)
+
+## Intended Audience
+
+This document is intended for front end developers who want to contribute to our user inteface. To get the most benefit from this guidance, you should have an understanding of the following technologies:
+
+- HTML
+- CSS
+- JavaScript
+	- D3 library
+	- ReactJS+Flux arquitecture
+	- NPM package manager
+
+## Getting Started
+
+Here you will find useful information to get you started on contributing to our User Interface (UI). For more information on how to run and use "Open Network Insight Operational Analytics User Interface" (ONI OA UI) please go to our [User Guide](https://github.com/Open-Network-Insight/open-network-insight/wiki/User%20Guide)
+
+## Technical Documentation
+
+There are two architectural approaches when it comes to the UI:
+
+1. [Flows: Inline JavaScript](#flows-inline-javascript)
+2. [DNS: ReactJS + Flux](#dns-reactjs--flux)
+
+### Flows: Inline JavaScript
+
+For more information please go [here](../flow/UI.md#flows-inline-javascript)
+
+### DNS: ReactJS + Flux
+
+The main issue with flows is that is hard to maintain. This pipeline introduces a new arquitecture that allows for modular code that is also scalable and easy to maintain.
+
+We have chosen ReactJS to build our UI and on top of that we are following the Flux arquitecture as a complement. This allows contributors to get on board quickly as they will find lots of information on the internet.
+
+For more information about ReactJS and Flux, please go to:
+
+- https://facebook.github.io/react/
+- https://facebook.github.io/flux/
+
+From now on we assume you are familiar with ReactJS+Flux applications.
+
+DNS has the following sections:
+
+1. [Dashboard](#dashboard)
+2. [Threat Investigation](#threat-investigation)
+3. [Storyboard](#storyboard)
+
+#### Dashboard
```

Comment:
A sentence or two clarifying what this is, what it does and what the bullets on the list are would be helpful.